rllib/examples/centralized_critic_2.py

```diff
@@ -0,0 +1,158 @@
+
+        self.value_model = FullyConnectedNetwork(obs_space, action_space, 1,
+                                                 model_config, name + "_vf")
+        self.register_variables(self.value_model.variables())
+
+    def forward(self, input_dict, state, seq_lens):
+        self._value_out, _ = self.value_model({
+            "obs": input_dict["obs_flat"]
+        }, state, seq_lens)
+        return self.action_model({
+            "obs": input_dict["obs"]["own_obs"]
+        }, state, seq_lens)
+
+    def value_function(self):
+        return tf.reshape(self._value_out, [-1])
+
+
+class GlobalObsTwoStepGame(MultiAgentEnv):
+    action_space = Discrete(2)
+    observation_space = Dict({
+        "own_obs": Discrete(6),
+        "opponent_obs": Discrete(6),
+        "opponent_action": Discrete(2),
+    })
+
+    def __init__(self, env_config):
+        self.env = TwoStepGame(env_config)
+
+    def reset(self):
+        obs_dict = self.env.reset()
+        return self.to_global_obs(obs_dict)
+
+    def step(self, action_dict):
+        obs_dict, rewards, dones, infos = self.env.step(action_dict)
+        return self.to_global_obs(obs_dict), rewards, dones, infos
+
+    def to_global_obs(self, obs_dict):
+        return {
+            self.env.agent_1: {
+                "own_obs": obs_dict[self.env.agent_1],
+                "opponent_obs": obs_dict[self.env.agent_2],
+                "opponent_action": 0,  # populated by fill_in_actions
+            },
+            self.env.agent_2: {
+                "own_obs": obs_dict[self.env.agent_2],
+                "opponent_obs": obs_dict[self.env.agent_1],
+                "opponent_action": 0,  # populated by fill_in_actions
+            },
+        }
+
+
+def fill_in_actions(info):
+    """Callback that saves opponent actions into the agent obs.
+
+    If you don't care about opponent actions you can leave this out."""
+
+    to_update = info["post_batch"][SampleBatch.CUR_OBS]
+    my_id = info["agent_id"]
+    other_id = 1 if my_id == 0 else 0
+    action_encoder = ModelCatalog.get_preprocessor_for_space(Discrete(2))
+
+    # set the opponent actions into the observation
+    _, opponent_batch = info["all_pre_batches"][other_id]
+    opponent_actions = np.array([
+        action_encoder.transform(a)
+        for a in opponent_batch[SampleBatch.ACTIONS]
+    ])
+    to_update[:, -2:] = opponent_actions
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+    ModelCatalog.register_custom_model("cc_model", CentralizedCriticModel)
+    tune.run(
+        "PPO",
+        stop={
+            "timesteps_total": args.stop,
+            "episode_reward_mean": 7.99,
+        },
+        config={
+            "env": GlobalObsTwoStepGame,
+            "batch_mode": "complete_episodes",
+            "callbacks": {
+                "on_postprocess_traj": tune.function(fill_in_actions),
+            },
+            "num_workers": 0,
+            "multiagent": {
```

Comment:
This nested dict seems very clumsy; I wonder if somehow you can just sugar it (the `policies` and `policy_mapping_fn`).